test/unit/org/apache/cassandra/index/sai/memory/TrieMemtableIndexTest.java

```diff
@@ -18,158 +18,55 @@
-import org.apache.cassandra.db.ColumnFamilyStore;
 import org.apache.cassandra.db.DecoratedKey;
 import org.apache.cassandra.db.PartitionPosition;
 import org.apache.cassandra.db.marshal.Int32Type;
-import org.apache.cassandra.db.memtable.AbstractAllocatorMemtable;
 import org.apache.cassandra.db.memtable.TrieMemtable;
 import org.apache.cassandra.dht.AbstractBounds;
-import org.apache.cassandra.dht.BootStrapper;
 import org.apache.cassandra.dht.Bounds;
 import org.apache.cassandra.dht.ExcludingBounds;
-import org.apache.cassandra.dht.IPartitioner;
 import org.apache.cassandra.dht.IncludingExcludingBounds;
 import org.apache.cassandra.dht.Range;
-import org.apache.cassandra.index.sai.IndexContext;
 import org.apache.cassandra.index.sai.QueryContext;
-import org.apache.cassandra.index.sai.SAITester;
 import org.apache.cassandra.index.sai.plan.Expression;
 import org.apache.cassandra.index.sai.utils.PrimaryKey;
 import org.apache.cassandra.index.sai.utils.RangeIterator;
 import org.apache.cassandra.index.sai.utils.TypeUtil;
-import org.apache.cassandra.inject.Injections;
-import org.apache.cassandra.inject.InvokePointBuilder;
 import org.apache.cassandra.io.compress.BufferType;
-import org.apache.cassandra.locator.TokenMetadata;
-import org.apache.cassandra.schema.MockSchema;
-import org.apache.cassandra.schema.TableMetadata;
-import org.apache.cassandra.service.StorageService;
-import org.apache.cassandra.utils.FBUtilities;
 import org.apache.cassandra.utils.Pair;
-import org.apache.cassandra.utils.ReflectionUtils;
 import org.apache.cassandra.utils.bytecomparable.ByteComparable;
 import org.apache.cassandra.utils.bytecomparable.ByteSource;
-import org.apache.cassandra.utils.concurrent.OpOrder;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
 
-public class TrieMemtableIndexTest extends SAITester
+public class TrieMemtableIndexTest extends TrieMemtableIndexTestBase
 {
-    private static final Injections.Counter indexSearchCounter = Injections.newCounter("IndexSearchCounter")
-                                                                           .add(InvokePointBuilder.newInvokePoint()
-                                                                                                  .onClass(TrieMemoryIndex.class)
-                                                                                                  .onMethod("search"))
-                                                                           .build();
-
-    private ColumnFamilyStore cfs;
-    private IndexContext indexContext;
-    private TrieMemtableIndex memtableIndex;
-    private AbstractAllocatorMemtable memtable;
-    private IPartitioner partitioner;
-    private Map<DecoratedKey, Integer> keyMap;
-    private Map<Integer, Integer> rowMap;
-
     @BeforeClass
     public static void setShardCount()
     {
         System.setProperty("cassandra.trie.memtable.shard.count", "8");
```

Comment:
If we are not running this test with multiple memory access methods specified, I would explicitly set the allocation type here as well.

Reply:
We weren't but as we talked today - we better do. I will improve the tests